homeassistant/components/zwave_me/__init__.py

```diff
@@ -0,0 +1,102 @@
+"""The Z-Wave-Me WS integration."""
+import logging
+
+from zwave_me_ws import ZWaveMe, ZWaveMeData
+
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import HomeAssistant, callback
+from homeassistant.helpers.dispatcher import async_dispatcher_connect, dispatcher_send
+from homeassistant.helpers.entity import Entity
+
+from .const import DOMAIN, PLATFORMS, ZWAVE_PLATFORMS
+
+_LOGGER = logging.getLogger(__name__)
+
+
+async def async_setup_entry(hass, entry):
+    """Set up Z-Wave-Me from a config entry."""
+    hass.data[DOMAIN] = {}
+    hass.data[DOMAIN][entry.entry_id] = ZWaveMeController(hass, entry)
+    if await hass.data[DOMAIN][entry.entry_id].async_establish_connection():
+        hass.config_entries.async_setup_platforms(entry, PLATFORMS)
+        hass.data[DOMAIN][entry.entry_id].zwave_api.get_devices()
```

Comment:
We need to move out the platform setup and getting the devices to a separate coroutine function so that we can gather and await the platform forward. We don't want to get the devices until the platforms are setup. But we can't await the platform setup from `async_setup_entry` since that may deadlock.

```py
async async_setup_platforms(
    hass: HomeAssistant, entry: ConfigEntry, controller: ZWaveMeController
) -> None:
    """Set up platforms."""
    await asyncio.gather(
        *[
            hass.config_entries.async_forward_entry_setup(entry, platform)
            for platform in PLATFORMS
        ]
    )
    await hass.async_add_executor_job(controller.zwave_api.get_devices)
```